agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/api/resource/ebs_discovery.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	ebsVolumeDiscoveryTimeout = 5 * time.Second
+	ebsResourceKeyPrefix      = "ebs-volume:"
+	ScanPeriod                = 500 * time.Millisecond
+)
+
+var (
+	ErrInvalidVolumeID = errors.New("EBS volume IDs do not match")
+)
+
+type EBSDiscoveryClient struct {
+	ctx context.Context
+}
+
+func NewDiscoveryClient(ctx context.Context) *EBSDiscoveryClient {
+	return &EBSDiscoveryClient{
+		ctx: ctx,
+	}
+}
+
+// ScanEBSVolumes will iterate through the entire list of pending EBS volume attachments within the agent state and checks if it's attached on the host.
```

Comment:
"pending EBS volume attachments within the agent state" is a little confusing without the additional context that `pendingAttachments` comes from agent state. How about just saying that "entire list of provided EBS attachments"?